add coc-diagnostic-first and coc-diagnostic-first-error

kovetskiy opened the pull request:
Add two more Plug commands for a case when you want to build code & jump
to the first found error.

kovetskiy commented on plugin/coc.vim:
```diff
@@ -289,34 +289,36 @@ command! -nargs=* -bar -complete=custom,s:InstallOptions CocInstall   :call coc#
+nnoremap <Plug>(coc-diagnostic-first)       :<C-u>call       CocActionAsync('diagnosticFirst')<CR>
+nnoremap <Plug>(coc-diagnostic-next-error)  :<C-u>call       CocActionAsync('diagnosticNext',     'error')<CR>
+nnoremap <Plug>(coc-diagnostic-prev-error)  :<C-u>call       CocActionAsync('diagnosticPrevious', 'error')<CR>
+nnoremap <Plug>(coc-diagnostic-first-error) :<C-u>call       CocActionAsync('diagnosticFirst',    'error')<CR>
```
Unfortunately, I had to add spaces to all other mappings because `nnoremap <Plug>(coc-diagnostic-first-error)` was too long.

kovetskiy replied:
Please use `w=1`  to see the differences `https://github.com/neoclide/coc.nvim/pull/1328/files?w=1`

fannheyward commented on src/diagnostic/manager.ts:
```diff
@@ -323,6 +323,19 @@ export class DiagnosticManager implements Disposable {
+    if (ranges.length == 0) {
+      return
+    }
+    await this.jumpTo(ranges[0])
```
There is no `jumpTo` in the origin master.

chemzqm commented:
@kovetskiy I don't want to list too many `nnoremap` key-mappings for diagnostics jump, simply  extend `CocAction` should be enough.

kovetskiy commented:
@fannheyward Thanks for the review, I definitely had my upstream master out-dated, I've fixed it.

@chemzqm Thanks for the review, I've removed the <Plug> nnoremaps.

I made force push to keep it clean.
